Integrations/python/deephaven/transferrer/__init__.py

```diff
@@ -61,21 +61,29 @@ def table_to_numpy_2d(idx, cols, np_dtype = None):
         np_dtype = np.intc
 
     if np_dtype == np.bool_:
-        buffer = gatherer.booleanTensorBuffer2D(idx, cols)
+        buffer = _gatherer.booleanTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.byte:
-        buffer = gatherer.byteTensorBuffer2D(idx, cols)
+        buffer = _gatherer.byteTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.double:
-        buffer = gatherer.doubleTensorBuffer2D(idx, cols)
+        buffer = _gatherer.doubleTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.intc:
-        buffer = gatherer.intTensorBuffer2D(idx, cols)
+        buffer = _gatherer.intTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.int_:
-        buffer = gatherer.longTensorBuffer2D(idx, cols)
+        buffer = _gatherer.longTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.short:
-        buffer = gatherer.shortTensorBuffer2D(idx, cols)
+        buffer = _gatherer.shortTensorBuffer2D(idx, cols, transpose)
     elif np_dtype == np.single:
-        buffer = gatherer.floatTensorBuffer2D(idx, cols)
+        buffer = _gatherer.floatTensorBuffer2D(idx, cols, transpose)
+    else:
+        raise ValueError("Data type {input_type} is not supported.".format(input_type = np_dtype))
 
     tensor = np.frombuffer(buffer, dtype = np_dtype)
-    tensor.shape = (idx.getSize(), len(cols))
+    if not transpose:
+        tensor.shape = (idx.getSize(), len(cols))
+    else:
+        tensor.shape = (len(cols), idx.getSize())
 
-    return np.squeeze(tensor)
\ No newline at end of file
+    if squeeze:
```

Comment:
It is good that this is optional.  Does it make sense to even be here?  A user could easily do something like the following without increasing the complexity of the code.  
```
tensor = np.squeeze(table_to_numpy_2d(...))
```
Having the option increases the complexity of the functionality.  It also causes a subtle bit of confusion.  The method has 2D in the name.  Without squeeze, we get a 2D tensor.  If you use the squeeze flag, now you get a 1D tensor.  Now the docs are not right.

Did the benefit of the argument make up for the problems?